config/factory.go

```diff
@@ -101,6 +82,61 @@ func EmptyConfig() MutableNodeConfig {
 	}
 }
 
+func DefaultConfig() MutableNodeConfig {
+	cfg := EmptyConfig()
+
+	cfg.SetActiveConsensusAlgo(consensus.CONSENSUS_ALGO_TYPE_BENCHMARK_CONSENSUS)
+
+	cfg.SetUint32(VIRTUAL_CHAIN_ID, 42)
+
+	cfg.SetUint32(BLOCK_TRACKER_GRACE_DISTANCE, 3)
+
+	cfg.SetDuration(BLOCK_SYNC_BATCH_SIZE, 10000)
+	cfg.SetDuration(BLOCK_SYNC_INTERVAL, 5*time.Second)
+	cfg.SetDuration(BLOCK_SYNC_COLLECT_RESPONSE_TIMEOUT, 3*time.Second)
+	cfg.SetDuration(BLOCK_SYNC_COLLECT_CHUNKS_TIMEOUT, 5*time.Second)
+
+	cfg.SetDuration(BLOCK_TRANSACTION_RECEIPT_QUERY_GRACE_START, 5*time.Second)
+	cfg.SetDuration(BLOCK_TRANSACTION_RECEIPT_QUERY_GRACE_END, 5*time.Second)
+	cfg.SetDuration(BLOCK_TRANSACTION_RECEIPT_QUERY_EXPIRATION_WINDOW, 3*time.Minute)
+
+	cfg.SetUint32(STATE_STORAGE_HISTORY_RETENTION_DISTANCE, 5)
+
+	cfg.SetUint32(CONSENSUS_CONTEXT_MINIMUM_TRANSACTION_IN_BLOCK, 1)
+	cfg.SetUint32(CONSENSUS_CONTEXT_MAXIMUM_TRANSACTION_IN_BLOCK, 1)
+
+	cfg.SetUint32(TRANSACTION_POOL_PENDING_POOL_SIZE_IN_BYTES, 20*1024*1024)
+	cfg.SetDuration(TRANSACTION_POOL_TRANSACTION_EXPIRATION_WINDOW, 30*time.Minute)
+	cfg.SetDuration(TRANSACTION_POOL_PENDING_POOL_CLEAR_EXPIRED_INTERVAL, 10*time.Second)
+	cfg.SetDuration(TRANSACTION_POOL_COMMITTED_POOL_CLEAR_EXPIRED_INTERVAL, 30*time.Second)
+
+	cfg.SetUint32(GOSSIP_LISTEN_PORT, 4400)
+
+	cfg.SetString(PROCESSOR_ARTIFACT_PATH, filepath.Join(GetProjectSourceTmpPath(), "processor-artifacts"))
+
+	return cfg
+}
+
+func (c *config) OverrideNodeSpecificValues(
```

Comment:
This method signature is confusing. When you return an argument, I assume the method has no side effects (is referentially transparent), which isn't the case. Either make this a copying method (leaving the original receiver unchanged, or change the signature to a void